mlflow/models/evaluation/base.py

```diff
@@ -151,6 +153,93 @@ def __str__(self):
         return "EvaluationMetric(" + ", ".join(parts) + ")"
 
 
+class GenAIEvaluationMetric(EvaluationMetric):
```

Comment:
Note: What if we copy the class def from the function and override the __call__ on the copy before instantiation? 
(Notes from standup discussion)